control-plane/build-support/functions/10-util.sh

```diff
@@ -2,984 +2,932 @@
-   return $?
+	# Arguments:
+	#   $1 - Path to the version file
+	#   $2 - Version string
+	#   $3 - PreRelease version (if unset will become an empty string)
+	#
+	# Returns:
+	#   0 - success
+	#   * - error
+
+	if ! test -f "$1"; then
+		err "ERROR: '$1' is not a regular file. update_version must be called with the path to a go version file"
+		return 1
+	fi
+
+	if test -z "$2"; then
+		err "ERROR: The version specified was empty"
+		return 1
+	fi
+
+	local vfile="$1"
+	local version="$2"
+	local prerelease="$3"
+
+	sed_i ${SED_EXT} -e "s/(Version[[:space:]]*=[[:space:]]*)\"[^\"]*\"/\1\"${version}\"/g" -e "s/(VersionPrerelease[[:space:]]*=[[:space:]]*)\"[^\"]*\"/\1\"${prerelease}\"/g" "${vfile}"
+	return $?
 }
 
 function update_version_helm {
-   # Arguments:
-   #   $1 - Path to the directory where the root of the Helm chart is
-   #   $2 - Version string
-   #   $3 - PreRelease version (if unset will become an empty string)
-   #   $4 - Image base path
-   #
-   # Returns:
-   #   0 - success
-   #   * - error
-
-   if ! test -d "$1"
-   then
-      err "ERROR: '$1' is not a directory. update_version_helm must be called with the path to the Helm chart"
-      return 1
-   fi
-
-   if test -z "$2"
-   then
-      err "ERROR: The version specified was empty"
-      return 1
-   fi
-
-   local vfile="$1/values.yaml"
-   local cfile="$1/Chart.yaml"
-   local version="$2"
-   local prerelease="$3"
-   local full_version="$2"
-   if ! test -z "$3"
-   then
-     full_version="$2-$3"
-   fi
-
-   sed_i ${SED_EXT} -e "s/(imageK8S:.*\/consul-k8s-control-plane:)[^\"]*/imageK8S: $4${full_version}/g" "${vfile}"
-   sed_i ${SED_EXT} -e "s/(version:[[:space:]]*)[^\"]*/\1${full_version}/g" "${cfile}"
-   sed_i ${SED_EXT} -e "s/(image:.*\/consul-k8s-control-plane:)[^\"]*/image: $4${full_version}/g" "${cfile}"
-
-   if test -z "$3"
-   then
-     sed_i ${SED_EXT} -e "s/(artifacthub.io\/prerelease:[[:space:]]*)[^\"]*/\1false/g" "${cfile}"
-   else
-     sed_i ${SED_EXT} -e "s/(artifacthub.io\/prerelease:[[:space:]]*)[^\"]*/\1true/g" "${cfile}"
-   fi
-   return $?
+	# Arguments:
+	#   $1 - Path to the directory where the root of the Helm chart is
+	#   $2 - Version string
+	#   $3 - Release version (if unset will become an empty string)
+	#   $4 - Image base path
+	#   $5 - Consul version string
+	#   $6 - Consul image base path
+	#
+	# Returns:
+	#   0 - success
+	#   * - error
+
+	if ! test -d "$1"; then
+		err "ERROR: '$1' is not a directory. update_version_helm must be called with the path to the Helm chart"
+		return 1
+	fi
+
+	if test -z "$2"; then
+		err "ERROR: The version specified was empty"
+		return 1
+	fi
+
+	local vfile="$1/values.yaml"
+	local cfile="$1/Chart.yaml"
+	local version="$2"
+	local consul_version="$5"
+	local prerelease="$3"
+	local full_version="$2"
+	local full_consul_version="$5"
```

Comment:
This function has a double use case and is used by both the `make prepare-dev` and `make prepare-release` targets.

$3 being "dev" acts as a toggle when this function is called. If those variables are unset then it will be incorrect when it is called by `make prepare-release`